.github/workflows/build-artifact.yml

```diff
@@ -33,12 +33,24 @@ jobs:
         sha=$(echo ${{ github.sha }} | cut -c 1-8)
         echo "::set-env name=artifact_sha::$sha"
 
+    - name: Check dependencies cache
+      id: cache-dependencies
+      uses: actions/cache@v2
+      with:
+        path: ./plugin/ThirdParty
+        key: ${{ runner.os }}-cactbot-${{ hashFiles('./util/fetch_deps.py', './util/DEPS.py') }}
+        restore-keys: |
```

Comment:
I'm not quite sure I understand what `restore-keys` is doing here.  Can you explain?

Reply:
When cache `key` fully match the `key`property here, then its `outputs.cache-hit` would be `true`, so step "setup python" and "fetch dependency" would be skipped.

When no any cache `key` fully match the `key` property, then its `outputs.cache-hit` would be always `false`. But the most recent cache would be recovered if any `restore-keys` match its prefix.

Maybe this feature is not quite useful for `plugin` building here, but as for `npm` or `pip`, even the cache key was not exactly the same, but there might be some libraries already in that cache, so when running `npm install` or `pip install`, those libraries would be skipped, a little faster, right?

Reply:
For example in [this run](https://github.com/MaikoTan/cactbot/runs/909896093?check_suite_focus=true) it matched prefix only.

Reply:
I get it! Thanks for the example.